### DISCUSSION TITLE
在devcontainer镜像中加入 go language server tools

### DISCUSSION THREAD

**Discussion by liuhaoyang** [alibaba/ilogtail#435] [Help] [answered]
希望在开发镜像内置下面的 tools，不然每次创建新的devcontainer都需要手动安装
```
  gotests
  gomodifytags
  impl
  goplay
  dlv
  staticcheck
  gopls
  go-outline
```

**Comment by liuhaoyang** [comment]
同时我还遇到一个devcontainer的使用问题。我在linux主机使用非root用户启动了devcontainer，在开发容器里面不能访问 `/root/go` 和下面的子目录，导致手动安装上面的 tools 提示权限不足。
![image](https://user-images.githubusercontent.com/10701915/192517373-52b47ec8-7446-4d82-bb05-e6278f271cf5.png)

在容器中使用`sudo` 时，提升需要账户的密码，我使用主机相同名称账户的密码也无效
![image](https://user-images.githubusercontent.com/10701915/192517542-6b3edb71-35e9-4782-973e-5d5408daeb7d.png)

ps： 上面的问题对 c++ core 和 go 的插件编译都没有影响。只影响了 vsc 的 go 开发体验，代码提示和符号查找/跳转都无法使用

**Comment by yyuuttaaoo** [comment, marked as answer]
tools chain里面有，不过权限没搞对，所以vscode无法使用。现在fix了。https://github.com/alibaba/ilogtail/pull/437/files
